internal/pkg/daemon/profilerecorder/profilerecorder.go

```diff
@@ -360,70 +372,208 @@ func (r *RecorderReconciler) collectLocalProfiles(
+		return errors.Wrap(err, "create seccompProfile resource")
+	}
+
+	r.log.Info("Created/updated profile", "action", res, "name", profileName)
+	r.record.Event(
+		profile,
+		event.Normal(reasonProfileCreated, "seccomp profile created"),
+	)
+
+	// Reset the syscalls for further recordings
+	if _, err := r.grpcClient.ResetSyscalls(ctx, request); err != nil {
+		return errors.Wrapf(
+			err, "reset syscalls for profile %s", profileID,
 		)
+	}
 
-		// Reset the syscalls for further recordings
-		if _, err := r.grpcClient.ResetSyscalls(ctx, request); err != nil {
-			return errors.Wrapf(
-				err, "reset syscalls for profile %s", profileID,
-			)
-		}
+	return nil
+}
+
+func (r *RecorderReconciler) collectLogSelinuxProfile(
+	ctx context.Context,
+	profileName string,
+	namespace string,
+	profileID string,
+) error {
+	// Retrieve the syscalls for the recording
+	request := &api.AvcRequest{Profile: profileID}
+	response, err := r.grpcClient.Avcs(ctx, request)
+	if err != nil {
+		return errors.Wrapf(
+			err, "retrieve avcs for profile %s", profileID,
+		)
+	}
+
+	selinuxProfileSpec := selinuxv1lpha1.SelinuxProfileSpec{}
+
+	profile := &selinuxv1lpha1.SelinuxProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      profileName,
+			Namespace: namespace,
+		},
+		Spec: selinuxProfileSpec,
+	}
+
+	selinuxProfileSpec.Policy = r.formatSelinuxProfile(profile, response)
+	r.log.Info("Created", "profile", profile)
+
+	res, err := controllerutil.CreateOrUpdate(ctx, r.client, profile,
+		func() error {
+			profile.Spec = selinuxProfileSpec
+			return nil
+		},
+	)
+	if err != nil {
+		r.log.Error(err, "Cannot create selinuxprofile resource")
+		r.record.Event(profile, event.Warning(reasonProfileCreationFailed, err))
+		return errors.Wrap(err, "create selinuxprofile resource")
+	}
+	r.log.Info("Created/updated selinux profile", "action", res, "name", profileName)
+	r.record.Event(
+		profile,
+		event.Normal(reasonProfileCreated, "selinuxprofile profile created"),
+	)
+
+	// Reset the selinuxprofile for further recordings
+	if _, err := r.grpcClient.ResetAvcs(ctx, request); err != nil {
+		return errors.Wrapf(
+			err, "reset selinuxprofile for profile %s", profileName,
+		)
 	}
 
 	return nil
 }
 
+func (r *RecorderReconciler) formatSelinuxProfile(
+	selinuxprofile *selinuxv1lpha1.SelinuxProfile,
+	avcResponse *api.AvcResponse,
+) string {
+	permMap := make(map[string]sets.String)
+	recorderContext := selinuxRecorderContext()
+
+	for _, avc := range avcResponse.GetAvc() {
+		r.log.Info("Received an AVC response",
+			"perm", avc.Perm, "tclass",
+			avc.Tclass, "scontext", avc.Scontext,
+			"tcontext", avc.Tcontext)
+
+		key := avc.Tclass + " " + ctxt2type(avc.Tcontext)
+		perms, ok := permMap[key]
+		if ok {
+			perms.Insert(avc.Perm)
+		} else {
+			permMap[key] = sets.NewString(avc.Perm)
+		}
+	}
+
+	policyBuilder := strings.Builder{}
+	policyBuilder.WriteString("(blockinherit container)\n")
+	for key, val := range permMap {
```

Comment:
I will do this (it's WIP locally, but I'm not working tomorrow and wanted to at least fix some things so you can continue the review). There should already be no duplicates, because we use a Set() on the gRPC server side. But you have a good point about sorting.